### PR TITLE
move SchemaConverters to Java

### PR DESCRIPTION
**Pull request by Gaurangi94**

### CODE REVIEW

File: connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            return UTF8String.fromBytes(((Utf8)value).getBytes());
+        }
+
+        if (LegacySQLTypeName.BYTES.equals(field.getType())) {
+            return getBytes((ByteBuffer)value);
+        }
+
+        if (LegacySQLTypeName.NUMERIC.equals(field.getType())) {
+            byte[] bytes = getBytes((ByteBuffer)value);
+            BigDecimal b = new BigDecimal(new BigInteger(bytes), BQ_NUMERIC_SCALE);
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:85]: @davidrabinowitz is there a buffer leak possible here?

File: connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        buf.get(bytes);
+
+        return bytes;
+    }
+
+    // Schema is not recursive so add helper for sequence of fields
+    static GenericInternalRow convertAll(FieldList fieldList,
+                                  GenericRecord record,
+                                  List<String> namesInOrder) {
+
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:111]: remove extra space

File: connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndITSuite.scala [hunk 3]
```diff
@@ -301,7 +301,6 @@ class SparkBigQueryEndToEndITSuite extends FunSuite
     row should contain theSameElementsInOrderAs expectedRow
   }
 
-
```
Comment (Gaurangi94) [inline connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndITSuite.scala:304]: revert

File: connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java (hunk 1 above)
Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:87]: No, it should be ok.

File: connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.unsafe.types.UTF8String;
+
+public class SchemaConverters {
+    // Numeric is a fixed precision Decimal Type with 38 digits of precision and 9 digits of scale.
+    // See https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#numeric-type
+    private final static int BQ_NUMERIC_PRECISION = 38;
+    private final static int BQ_NUMERIC_SCALE = 9;
+    private final static DecimalType NUMERIC_SPARK_TYPE = DataTypes.createDecimalType(
```
Comment (davidrabinowitz) [inline connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:29]: Fix indentation

File: connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+    static Object convert(Field field, Object value) {
+        if (value == null) {
+            return null;
+        }
+
+        if (field.getMode() == Field.Mode.REPEATED) {
+            // rather than recurring down we strip off the repeated mode
+            // Due to serialization issues, reconstruct the type using reflection:
+            // See: https://github.com/googleapis/google-cloud-java/issues/3942
```
Comment (Gaurangi94) [inline connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:53]: @davidrabinowitz Scala used "Some" and "Option" annotations here. I am not sure how that would be applicable in Java

Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:53]: Originally they are used in the `convert(Field field)` method, not in this one - and they are not needed in Java. Do you have another place where those are used?

Comment (Gaurangi94) [reply connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:53]: Not in this file.

**Pull request merged**